models/oauth2_application.go

```diff
@@ -431,6 +435,36 @@ func (grant *OAuth2Grant) increaseCount(e Engine) error {
 	return nil
 }
 
+// ScopeContains returns true if the grant scope contains the specified scope
+func (grant *OAuth2Grant) ScopeContains(scope string) bool {
+	for _, currentScope := range strings.Split(grant.Scope, " ") {
+		if scope == currentScope {
+			return true
+		}
+	}
+	return false
+}
+
+// SetNonce updates the current nonce value of a grant
+func (grant *OAuth2Grant) SetNonce(nonce string) error {
+	return grant.setNonce(x, nonce)
+}
+
+func (grant *OAuth2Grant) setNonce(e Engine, nonce string) error {
+	newGrant := new(OAuth2Grant)
+	newGrant.Nonce = nonce
+	_, err := e.ID(grant.ID).Update(newGrant)
```

Comment:
Use `Cols("nonce")` to indicate only update `nonce` column is better than automatically detection.

Reply:
Thanks for the hint, I changed to `Cols("nonce").Update(newGrant)`.